integrationTests/node/heartbeatV2/heartbeatV2_test.go

```diff
@@ -0,0 +1,473 @@
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/interceptors"
+	interceptorFactory "github.com/ElrondNetwork/elrond-go/process/interceptors/factory"
+	interceptorsProcessor "github.com/ElrondNetwork/elrond-go/process/interceptors/processor"
+	processMock "github.com/ElrondNetwork/elrond-go/process/mock"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/testscommon/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/testscommon/p2pmocks"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	defaultNodeName           = "node"
+	timeBetweenPeerAuths      = 10 * time.Second
+	timeBetweenHeartbeats     = 2 * time.Second
+	timeBetweenSendsWhenError = time.Second
+	thresholdBetweenSends     = 0.2
+)
+
+func TestHeartbeatV2_AllPeersSendMessages(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	keyGen := signing.NewKeyGenerator(mcl.NewSuiteBLS12())
+	sigHandler := createMockPeerSignatureHandler(keyGen)
+
+	interactingNodes := 3
+	nodes, senders, dataPools := createAndStartNodes(interactingNodes, keyGen, sigHandler)
+	assert.Equal(t, interactingNodes, len(nodes))
+	assert.Equal(t, interactingNodes, len(senders))
+	assert.Equal(t, interactingNodes, len(dataPools))
+
+	// Wait for messages to broadcast
+	time.Sleep(time.Second * 5)
+
+	// Check sent messages
+	maxMessageAgeAllowed := time.Second * 7
+	checkMessages(t, nodes, dataPools, maxMessageAgeAllowed)
+
+	closeComponents(t, nodes, senders, dataPools, nil)
+}
+
+func TestHeartbeatV2_PeerJoiningLate(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	keyGen := signing.NewKeyGenerator(mcl.NewSuiteBLS12())
+	sigHandler := createMockPeerSignatureHandler(keyGen)
+	shardCoordinator := &sharding.OneShardCoordinator{}
+
+	interactingNodes := 3
+	nodes, senders, dataPools := createAndStartNodes(interactingNodes, keyGen, sigHandler)
+	assert.Equal(t, interactingNodes, len(nodes))
+	assert.Equal(t, interactingNodes, len(senders))
+	assert.Equal(t, interactingNodes, len(dataPools))
+
+	// Wait for messages to broadcast
+	time.Sleep(time.Second * 3)
+
+	// Check sent messages
+	maxMessageAgeAllowed := time.Second * 5
+	checkMessages(t, nodes, dataPools, maxMessageAgeAllowed)
+
+	// Add new delayed node which requests messages
+	delayedNode, delayedNodeDataPool := createDelayedNode(nodes, sigHandler)
+	nodes = append(nodes, delayedNode)
+	dataPools = append(dataPools, delayedNodeDataPool)
+
+	pksArray := make([][]byte, 0)
+	for _, node := range nodes {
+		pksArray = append(pksArray, node.ID().Bytes())
+	}
+
+	// Create resolvers and request chunk from delayed node
+	paResolvers := createPeerAuthResolvers(pksArray, nodes, dataPools, shardCoordinator)
+	newNodeIndex := len(nodes) - 1
+	_ = paResolvers[newNodeIndex].RequestDataFromChunk(0, 0)
```

Comment:
this can be replaced with the request processor as to test that component also. Can add a TODO so we can do it later.

Reply:
done